pkg/capi/edit/turtles-capi.cattle.io.capiprovider/index.vue

```diff
@@ -99,24 +99,38 @@ export default defineComponent({
   },
 
   methods: {
-
     clickedType(obj: ProviderType) {
       const id = obj.id;
 
-      this.$router.applyQuery({ [SUB_TYPE]: id });
+      this.$router?.applyQuery({ [SUB_TYPE]: id });
       this.selectType(id);
     },
 
     selectType(type: string) {
       this.subType = type;
       this.$emit('set-subtype', this.$store.getters['i18n/withFallback'](`cluster.provider."${ type }"`, null, type));
     },
+    handleValueChanged(newVal: {k: string, val: any}) {
```

Comment:
This seems like a very generic thing to be defining in multiple components. It looks like either the set method from `@rancher/shell/utils/object` would work, or lodash `set`